Block syscall related to signal handling

pvollan opened the pull request:
#### 43c46e90c40092e5bcc033ef8be07be56f3401c1
<pre>
Block syscall related to signal handling
<a href="https://bugs.webkit.org/show_bug.cgi?id=246020">https://bugs.webkit.org/show_bug.cgi?id=246020</a>
&lt;rdar://100604972&gt;

Reviewed by Youenn Fablet.

The syscall SYS_sigreturn was unblocked in &lt;<a href="https://commits.webkit.org/255036@main">https://commits.webkit.org/255036@main</a>&gt;, but it turns out
it is sufficient to block it without the default process termination behavior.

Test: WebKit.SyscallUnixSandboxCheck

* Source/WTF/wtf/spi/darwin/SandboxSPI.h:
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::hasSandboxMachLookupAccessToXPCServiceName):
(WebCore::Internals::hasSandboxUnixSyscallAccess):
* Source/WebCore/testing/Internals.cpp.orig: Copied from Source/WebCore/testing/Internals.cpp.
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:
* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKit/SyscallUnixSandboxCheck.mm: Added.
(TEST):

Canonical link: <a href="https://commits.webkit.org/255462@main">https://commits.webkit.org/255462@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/8991150708aaf63b216791f7737792963e5a97bb

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/92598 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/1817 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/23183 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/102331 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/162735 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/96599 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/1815 "Built successfully") | [✅ 🛠 mac-debug](https://ews-build.webkit.org/#/builders/71/builds/30176 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/84997 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/98494 "Built successfully") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/98261 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/78/builds/1221 "Passed tests") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/79094 "Built successfully") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/28153 "Passed tests") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/9/builds/83139 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/3/builds/82838 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/71244 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/83967 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/66/builds/36585 "Built successfully") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/73/builds/16765 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/79009 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/67/builds/34372 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/17944 "Passed tests") | [  ~~🧪 jsc-armv7-tests~~](https://ews-build.webkit.org/#/builders/46/builds/27394 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/3788 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/63/builds/38242 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/60/builds/40558 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/81629 "Built successfully") | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/65/builds/40151 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/37103 "Passed tests") | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/18462 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

youennf commented on Source/WebCore/testing/Internals.cpp:
```diff
@@ -6402,6 +6402,23 @@ bool Internals::hasSandboxMachLookupAccessToXPCServiceName(const String& process
+    if (process == "com.apple.WebKit.WebContent"_s)
+        pid = getpid();
+    else
+        RELEASE_ASSERT_NOT_REACHED();
```
How about:
RELEASE_ASSERT(process == "com.apple.WebKit.WebContent"_s);
auto pid = getpid();

youennf commented on Source/WebCore/testing/Internals.h:
```diff
@@ -1237,7 +1237,8 @@ class Internals final : public RefCounted<Internals>, private ContextDestruction
     bool hasSandboxMachLookupAccessToXPCServiceName(const String& process, const String& service);
     bool hasSandboxIOKitOpenAccessToClass(const String& process, const String& ioKitClass);
-
+    bool hasSandboxUnixSyscallAccess(const String& process, unsigned syscall);
```
const?

pvollan replied:
Fixed!

Thanks for reviewing!

pvollan committed:
Block syscall related to signal handling
https://bugs.webkit.org/show_bug.cgi?id=246020
<rdar://100604972>

Reviewed by Youenn Fablet.

The syscall SYS_sigreturn was unblocked in <https://commits.webkit.org/255036@main>, but it turns out
it is sufficient to block it without the default process termination behavior.

Test: WebKit.SyscallUnixSandboxCheck

* Source/WTF/wtf/spi/darwin/SandboxSPI.h:
* Source/WebCore/testing/Internals.cpp:
(WebCore::Internals::hasSandboxMachLookupAccessToXPCServiceName):
(WebCore::Internals::hasSandboxUnixSyscallAccess):
* Source/WebCore/testing/Internals.cpp.orig: Copied from Source/WebCore/testing/Internals.cpp.
* Source/WebCore/testing/Internals.h:
* Source/WebCore/testing/Internals.idl:
* Source/WebKit/Resources/SandboxProfiles/ios/com.apple.WebKit.WebContent.sb.in:
* Source/WebKit/WebProcess/com.apple.WebProcess.sb.in:
* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKit/SyscallUnixSandboxCheck.mm: Added.
(TEST):

Canonical link: https://commits.webkit.org/255462@main
